src/test/v2.0/advanced/channels-support-2.0.ts

```diff
@@ -124,26 +128,30 @@ export class ChannelControl2_0 implements ChannelControl<Channel, Context> {
 
   setupAndValidateListener2 = async (channel: Channel, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void) => {
     if (channel) {
-      listener2 = await channel.addContextListener(null, async (context) => {
-        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+      listener2 = await channel.addContextListener(expectedContextType, (context) => {
+        if (expectedContextType != null) {
+          expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        }
         onComplete(context);
       });
     } else {
-      listener2 = await fdc3.addContextListener(null, async (context) => {
-        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+      listener2 = await fdc3.addContextListener(expectedContextType, (context) => {
+        if (expectedContextType != null) {
+          expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        }
         onComplete(context);
       });
     }
 
     validateListenerObject(listener2);
   }
 
-  setupContextChecker = async (channel: Channel, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void) => {
+  setupContextChecker = async (channel: Channel,  requestedContextType: string, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): Promise<void> => {
     //Retrieve current context from channel
-    await channel.getCurrentContext().then(async (context) => {
-      expect(context.type).to.be.equals(expectedContextType, errorMessage);
-      onComplete(context);
-    });
+    const context = (requestedContextType == undefined) ? await channel.getCurrentContext() : await channel.getCurrentContext(requestedContextType);
```

Comment:
see comments on 1.2 version of function
```suggestion
    const context = (requestedContextType == null) ? await channel.getCurrentContext() : await channel.getCurrentContext(requestedContextType);
```